lib/cryptozoologist/generator.rb

```diff
@@ -48,12 +48,49 @@ def random
 
     # Generates a string for a street address with a number and street. Only
     # uses animal dictionaries and does not respect config exclusions.
-    # 
+    #
     def street_address
       number = rand(1..9000)
       street = Dictionary.animals.sample
       street = street.split(" ").map! {|word| word.capitalize! }.join(" ")
       "#{number} #{street} #{Dictionary.addresses.sample}"
     end
+
+    # Generates a string for a U.S. state name which is partially replaced by
+    # alliterative words from other dictionaries (e.g. "Oregon" can become "Oregoose" or "Orabbit")
+    # Does not respect config exclusions
+    def state
+      libraries = [:animals, :clothing, :colors]
+      base_state_words = Dictionary.states.sample.split(" ")
+      has_two_words = base_state_words.length > 1
+      if has_two_words
+        replace_index = rand(0..1)
+        base_state_words[replace_index] = get_alliteration(libraries, base_state_words[replace_index][0]).capitalize
+        return base_state_words.join(" ")
+      else
+        base_word = base_state_words[0]
+        key_letter_index = rand(0...base_word.length-1)
+        insert_word = get_alliteration(libraries, base_word[key_letter_index]).downcase
+        if insert_word.length > base_word.length - key_letter_index && key_letter_index > 0
```

Comment:
for things like this that are the heart of the algorithm, a comment explaining what is happening is super helpful for me!